### ISSUE TITLE
Scratter file loading causes bad output with deeper structure

### ISSUE THREAD

**Issue by jscotka** [psss/fmf#18]
This example is very tricky and not sure if this way is good to show, because parental inheritance is easy to understand, but this defines same level in more files, and not easy to understand what happen and output is strange.

It is very hard to understand what happen there and why there are these values and priority of loadind these files is bad.

### Example data structure
```
09:22 $ tree .
.
├── a
│   ├── b
│   │   ├── c
│   │   │   └── main.fmf
│   │   └── main.fmf
│   └── main.fmf
└── main.fmf

3 directories, 4 files
```
```
09:23 $ cat main.fmf 
/a/b/c:
   x: 1
   first: True
```
```
09:23 $ cat a/main.fmf 
/b/c:
   x: 2
   second: True
```
```
09:23 $ cat a/b/main.fmf 
/c:
   x: 3
   third: True
```
```
09:23 $ cat a/b/c/main.fmf 
x: 4
fourth: True
```
### Actual output
```
09:23 $ fmf
xxx/a/b/c
fmf_config: ./a/main.fmf
second: True
x: 2
xxx/a/b/c
fmf_config: ./a/b/c/main.fmf
fourth: True
third: True
x: 4
xxx/a/b/c
first: True
fmf_config: ./main.fmf
x: 1
```
```
09:24 $ fmf --whole
xxx
fmf_config: ./main.fmf
xxx/a
fmf_config: ./a/main.fmf
xxx/a/b/c
fmf_config: ./a/main.fmf
second: True
x: 2
xxx/a/b
fmf_config: ./a/b/main.fmf
xxx/a/b/c
fmf_config: ./a/b/c/main.fmf
fourth: True
third: True
x: 4
xxx/a/b/c
first: True
fmf_config: ./main.fmf
x: 1
```

### Expected output
```
$ fmf
xxx/a/b/c
fmf_config: ./a/b/c/main.fmf
first: True
second: True
third: True
fourth: True
x: 4
```

**Comment by jscotka** [comment]
@jkrysl yes #17 is same issue.

**Comment by jscotka** [comment]
Ive created this issue, based on PR#15 done 6 days ago. Ive just removed it from description there and moved to separated issue

**Comment by psss** [comment]
This should now be fixed by 70b6207.
